crates/walrus-service/src/client/cli/runner.rs

```diff
@@ -1123,6 +1169,18 @@ impl ClientCommandRunner {
 
         internal_run_ctx.finalize_after_store(&mut store_args).await;
 
+        if let Some(collector) = tail_handle_collector {
+            let mut handles_guard = collector.lock().await;
+            let mut handles = Vec::new();
+            std::mem::swap(&mut *handles_guard, &mut handles);
+            drop(handles_guard);
+            for handle in handles {
+                if let Err(err) = handle.await {
+                    tracing::warn!(?err, "tail upload task failed");
+                }
+            }
+        }
```

Comment:
Question: does this mean that the tail handle is blocking?

Reply:
This is blocking only when detached mode is selected with `child_process_enabled` set to false. I can add a warning message in this particular scenario, so user doesn't have unrealistic expectations.